ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/executors/asynchronous.py

```diff
@@ -0,0 +1,75 @@
+"""
+Async executor based on ModelAPI
+"""
+# Copyright (C) 2021-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from typing import Any, Tuple, Union
+
+from openvino.model_zoo.model_api.pipelines import AsyncPipeline
+
+from ote_sdk.usecases.exportable_code.demo.demo_package.model_container import (
+    ModelContainer,
+)
+from ote_sdk.usecases.exportable_code.demo.demo_package.utils import (
+    create_output_converter,
+)
+from ote_sdk.usecases.exportable_code.streamer import get_streamer
+from ote_sdk.usecases.exportable_code.visualizers import HandlerVisualizer, Visualizer
+
+
+class AsyncExecutor:
+    """
+    Async inferencer
+
+    Args:
+        model: model for inference
+        visualizer: for visualize inference results
+    """
+
+    def __init__(self, model: ModelContainer, visualizer: Visualizer) -> None:
+        self.model = model.core_model
+        self.visualizer = visualizer
+        self.converter = create_output_converter(model.task_type, model.labels)
+        self.async_pipeline = AsyncPipeline(self.model)
+
+    def run(self, input_stream: Union[int, str], loop=False):
```

Comment:
```suggestion
    def run(self, input_stream: Union[int, str], loop: bool = False):
```

Please double-check everywhere.